optimism/src/keccak/interpreter.rs

```diff
@@ -791,19 +791,19 @@ pub trait KeccakInterpreter<F: One + Debug + Zero> {
     }
     /// Returns a variable that encodes whether the current step is an absorb sponge (1 = yes)
     fn is_absorb(&self) -> Self::Variable {
-        self.variable(KeccakColumn::FlagAbsorb)
+        self.variable(KeccakColumn::Selector(Absorb))
     }
     /// Returns a variable that encodes whether the current step is a squeeze sponge (1 = yes)
     fn is_squeeze(&self) -> Self::Variable {
-        self.variable(KeccakColumn::FlagSqueeze)
+        self.variable(KeccakColumn::Selector(Squeeze))
     }
     /// Returns a variable that encodes whether the current step is the first absorb sponge (1 = yes)
     fn is_root(&self) -> Self::Variable {
-        self.variable(KeccakColumn::FlagRoot)
+        self.variable(KeccakColumn::Selector(Root))
     }
-    /// Returns a degree-2 variable that encodes whether the current step is the last absorb sponge (1 = yes)
+    /// Returns a degree-1 variable that encodes whether the current step is the last absorb sponge (1 = yes)
     fn is_pad(&self) -> Self::Variable {
-        self.pad_length() * self.variable(KeccakColumn::InvPadLength)
+        self.variable(KeccakColumn::Selector(Pad))
```

Comment:
IIRC you added an extra selector here, right? Instead of storing length and inv_length

Reply:
Actually if the Last absorb and the Only absorb have their own selectors for themselves, then there's no need of storing the inverse lengths anymore to check whether it is nonzero.